src/payments/providers/stripe/stripe.provider.ts

```
@@ -84,6 +84,19 @@ export class StripeProvider implements PaymentProvider {
 
+  /**
+   * Checkout-created PaymentIntents often carry the session id in payment_details.order_reference.
+   * Metadata may be empty on the PaymentIntent while our initiation row only has provider_session_id.
+   */
+  private checkoutSessionIdFromPaymentIntent(pi: Stripe.PaymentIntent): string {
+    const ref = (
+      pi as Stripe.PaymentIntent & {
+        payment_details?: { order_reference?: string | null };
+      }
+    ).payment_details?.order_reference;
+    return typeof ref === 'string' && ref.startsWith('cs_') ? ref : '';
```

ESLint

[error] Replace `'string'·&&·ref.startsWith('cs_')·?·ref·:·''` with `"string"·&&·ref.startsWith("cs_")·?·ref·:·""`

(prettier/prettier)